### PR TITLE
28 create login page

### LINKED ISSUE #28: Create Login Page
Create a Login Layout that should load when the user opens the app.  
Should contain: 

- Login Button
- Empty Link for forgotten password
- Empty Link for New User
- Username and password fields

### PR DESCRIPTION
**Pull request by sruduke**
Base implementation of login page and main home page (entry point for navigation).

### CODE REVIEW

File: app/src/main/java/com/example/happymeals/LoginScreen.java [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+
+import androidx.appcompat.app.AppCompatActivity;
+
+import android.os.Bundle;
+import android.view.View;
+import android.widget.Button;
+import android.widget.EditText;
+import android.widget.TextView;
+
+public class LoginScreen extends AppCompatActivity {
```
Comment (jeastgaard) [inline app/src/main/java/com/example/happymeals/LoginScreen.java:11]: If this is an activity then the class name should reflect that.

File: app/src/main/java/com/example/happymeals/LoginScreen.java [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_login_screen);
+
+        TextView forgetPassword = findViewById(R.id.forgot_password);
```
Comment (jeastgaard) [inline app/src/main/java/com/example/happymeals/LoginScreen.java:18]: Variable name "forgetPassword" should be "forgotPassoword" I think?

File: app/src/main/java/com/example/happymeals/LoginScreen.java [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+import android.os.Bundle;
+import android.view.View;
+import android.widget.Button;
+import android.widget.EditText;
+import android.widget.TextView;
+
+public class LoginScreen extends AppCompatActivity {
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
```
Comment (jeastgaard) [inline app/src/main/java/com/example/happymeals/LoginScreen.java:14]: Overall styling, make sure to add the spaces between <, {, (, etc.

File: app/src/main/java/com/example/happymeals/MainActivity.java [hunk 4]
```diff
@@ -2,13 +2,25 @@
...
 
 public class MainActivity extends AppCompatActivity {
 
     @Override
     protected void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
         setContentView(R.layout.activity_main);
+
+        ArrayAdapter<String> adapter = new MainActivityListAdapter(this, 0);
+        ListView options = findViewById(R.id.option_list);
```
Comment (jeastgaard) [inline app/src/main/java/com/example/happymeals/MainActivity.java:18]: Styling spacing

File: app/src/main/java/com/example/happymeals/MainActivityListAdapter.java [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+import android.widget.ArrayAdapter;
+import android.widget.ImageView;
+import android.widget.TextView;
+
+import androidx.annotation.NonNull;
+import androidx.annotation.Nullable;
+
+import java.util.ArrayList;
+
+public class MainActivityListAdapter extends ArrayAdapter<String> {
```
Comment (jeastgaard) [inline app/src/main/java/com/example/happymeals/MainActivityListAdapter.java]: MainActivityListAdapter, I think the name should reflect what the adapter is for. Such as optionsListAdapter etc.

File: app/src/main/java/com/example/happymeals/MainActivity.java (hunk 4 above)
Comment (jeastgaard) [inline app/src/main/java/com/example/happymeals/MainActivity.java:18]: Why are we using an adapter for the options?

File: app/src/main/java/com/example/happymeals/MainActivityListAdapter.java [hunk 6]
```diff
@@ -0,0 +1,40 @@
...
+
+    public MainActivityListAdapter(Context context, int resource) {
+        super(context, resource, new String[]{"Ingredient Storage", "Recipes", "Shopping List", "Meal Plan"});
+        this.context = context;
+    }
+
+    @NonNull
+    @Override
+    public View getView(int position, @Nullable View convertView, @NonNull ViewGroup parent) {
+        View view = convertView;
```
Comment (jeastgaard) [inline app/src/main/java/com/example/happymeals/MainActivityListAdapter.java]: This is redundant, should just use the convertView

**Review by jeastgaard — CHANGES_REQUESTED** [review]
Mostly styling that needs to be changed and possibly class names. I wonder about the need for an array adapter for the main list on the main menu.

**Review by kstark1 — APPROVED** [review]

**Review by jeastgaard — APPROVED** [review]

**Pull request merged**